### PR TITLE
[IOSP-647] Add documentation for Danger HealthKit rule

### PR DESCRIPTION
**Pull request by AliSoftware**
This is the documentation for the new Danger rule which will soon be added by https://github.com/babylonhealth/babylon-ios/pull/11523

### CODE REVIEW

**Review by NSMyself — APPROVED** [review]
Great work!

**Review by danspndl — APPROVED** [review]
👏

**Comment by AliSoftware** [comment]
PS: We should wait for https://github.com/babylonhealth/babylon-ios/pull/11523 to be merged first before merging this documentation PR (in case some changes are requested in the implementation)

File: Cookbook/Technical-Documents/DangerRules.md [hunk 1]
```diff
@@ -139,6 +145,29 @@ Note that this rule can have false positives in rare cases. Especially if you us
...
+
+* Is HealthKit enabled in the Entitlements files?
+* Are the 2 keys `NSHealthShareUsageDescription` and `NSHealthUpdateUsageDescription` both set in Info.plist?
+* Is `HEALTHKIT_INTEGRATION_AVAILABLE` flag turned on in Build Settings under `SWIFT_ACTIVE_COMPILATION_CONDITIONS`?
+* Is `HealthKit.framework` linked with the target?
+* Is `BabylonHealthKitIntegrationSDK.framework` linked with target?
+
+If the answer to these questions for a given target and build configuration are not all the same (not all `true` nor all `false`), this means that there is an inconsistency in the HealthKit setup for this target and build config.  
+In that case, for each target+config, the rule will report a failure, and also add a markdown table listing the state of the HealthKit setup (✅/❌) for each of those 5 setup places for that target+config.
+
```
Comment (AliSoftware) [inline Cookbook/Technical-Documents/DangerRules.md:169]:
```suggestion

> See also [this documentation in our private repo](https://github.com/babylonhealth/babylon-ios/blob/develop/Documentation/importing-healthkit/HowToCorrectlySetHealthKitPermissionsInAnAppWhenEnablingHKFunctionality.md) to learn how to configure HealthKit in our projects.

```

Link to @diegopetrucci 's doc – which is a pending PR too, see https://github.com/babylonhealth/babylon-ios/pull/11507

**Review by Daniel1of1 — APPROVED** [review]

**Pull request merged**